### PR TITLE
emacs-app-devel: update to 20241021

### PR DESCRIPTION
**Pull request by tsujp**
#### Description

This is a weird one I think since I have chosen to specifically only update emacs-app-devel which also has differing compilation flags (native compilation is now default for example).

As and when Emacs 30 gets released I imagine the Portfile will get simpler, I've been using this general structure since August rebuilding from Emacs' master every other week or so.

###### Tested on

macOS 14.5 23F79 arm64
Command Line Tools 15.3.0.0.1.1708646388

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL in commit message? 
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

File: editors/emacs/Portfile
Comment (reneeotten) [inline editors/emacs/Portfile:8]: please no unneeded whitespace changes

File: editors/emacs/Portfile
Comment (reneeotten) [inline editors/emacs/Portfile:127]: the `epoch` field can never decrease, it should stay at `5` (see the [guide](https://guide.macports.org/chunked/reference.html))

File: editors/emacs/Portfile
Comment (reneeotten) [inline editors/emacs/Portfile:158]: shouldn't this be the same as above; in other words `--with-native-compilation=no`?

File: editors/emacs/Portfile
Comment (reneeotten) [inline editors/emacs/Portfile:146]: I doubt the additions here follow the modeline with respect to alignment (i.e., multiple of four spaces).

File: editors/emacs/Portfile
Comment (reneeotten) [inline editors/emacs/Portfile:151]: what is the purpose of this comments; what would be a better place to put it? And if there is such a place why not do it now?

**Review by reneeotten — CHANGES_REQUESTED** [review]

File: editors/emacs/Portfile
Comment (tsujp) [reply editors/emacs/Portfile:127]: Whoops, I really should try find my MacPorts notes again.

Comment (tsujp) [reply editors/emacs/Portfile:127]: I'm going to bump it to `6` so it guaranteed invalidates prior versions.

File: editors/emacs/Portfile
Comment (tsujp) [reply editors/emacs/Portfile:158]: I wasn't sure whether it is necessary to add the entire exact text (I think I recall seeing patterns where others didn't elsewhere). Will change it to an exact match.

File: editors/emacs/Portfile
Comment (tsujp) [reply editors/emacs/Portfile:151]: Ah yes. Do you want this as a new variant option or always present by default? The linked comment:

```
** New configuration option '--disable-gc-mark-trace'.
This disables the GC mark trace buffer for about 5 % better garbage
collection performance.  Doing so may make it more difficult for Emacs
developers to help finding GC-related bugs that you run into, which is
why it the mark trace buffer is enabled by default.
```

File: editors/emacs/Portfile
Comment (tsujp) [reply editors/emacs/Portfile:146]: Didn't realise that applied after the first non-whitespace character on the line as there are prior lines in this very file which do not follow the convention. Perhaps an autoformatter is required? I corrected some other ones (not mine) as well as my ones, but there are others in this file which would be too tedious to do manually.

File: editors/emacs/Portfile
Comment (reneeotten) [reply editors/emacs/Portfile]: no, don't do that. You updates the version so there is absolutely no need to do so. The only reason to increase the "epoch" is when you want to downgrade the version or of the versioning has changed so that a newer version isn't recognized as newer by "vercomp".  

Please refer to the guide entry O linked.

File: editors/emacs/Portfile
Comment (reneeotten) [reply editors/emacs/Portfile:8]: fine... I guess "port lint" and I don't agree here then 😉

File: editors/emacs/Portfile
Comment (reneeotten) [reply editors/emacs/Portfile]: I think it does have to be exact, or at least it doesn't hurt and ic clearer.

File: editors/emacs/Portfile
Comment (tsujp) [reply editors/emacs/Portfile]: I was doing so just in-case as I had a brain fart and thought it also compared the commit hash (late and very tired). I did read the link you sent before acting hehe. I'll set it back to 5.

**Comment by tsujp** [comment]
@reneeotten I've updated the devel version to 20241021 master HEAD `b3af195213518514f78ac6f66f9598e45befd1ec`.

File: editors/emacs/Portfile
Comment (reneeotten) [inline editors/emacs/Portfile:161]: this could/should use `configure.args-replace` instead

File: editors/emacs/Portfile
Comment (reneeotten) [reply editors/emacs/Portfile:153]: well, it depends - do you think many users will be "Emacs developers" as that seems to be the usual situation where you might not want to have it disabled. I would probably have it use this configure.arg always; if @drkp has a different opinion he can always change it and/or put it in a variant.

File: editors/emacs/Portfile
Comment (tsujp) [reply editors/emacs/Portfile:161]: Will get to it, going to wait for @drkp's feedback for the other item in this file (do both at once).

Comment (reneeotten) [reply editors/emacs/Portfile:161]: Just finish it up. None of the maintainers had responded in over three weeks, so don't wait. Just get these last things finalized and I'll merge it.

Comment (tsujp) [reply editors/emacs/Portfile:161]: Alrighty, will get on it in an hour or two.

File: editors/emacs/Portfile
Comment (drkp) [reply editors/emacs/Portfile:153]: agree - I think we should leave this disabled by default, and if someone really wants it enabled for development we can create a variant. (Or if they're emacs developers, they can probably figure it out themselves. :-)

**Comment by drkp** [comment]
Looks good to me - do you still have some changes to make? We can merge when they're done.

**Comment by tsujp** [comment]
> Looks good to me - do you still have some changes to make? We can merge when they're done.

Pushing final changes now.

File: editors/emacs/Portfile
Comment (tsujp) [reply editors/emacs/Portfile:153]: I will leave it as-is and not create a variant then.

**Pull request merged**
